src/main/java/tools/jackson/core/io/IOContext.java

```diff
@@ -104,16 +107,21 @@ public class IOContext
      * Main constructor to use.
      *
      * @param streamReadConstraints constraints for streaming reads
+     * @param streamWriteConstraints constraints for streaming writes
      * @param br BufferRecycler to use, if any ({@code null} if none)
      * @param contentRef Input source reference for location reporting
      * @param managedResource Whether input source is managed (owned) by Jackson library
      * @param enc Encoding in use
      */
-    public IOContext(StreamReadConstraints streamReadConstraints, BufferRecycler br,
-            ContentReference contentRef, boolean managedResource,
+    public IOContext(StreamReadConstraints streamReadConstraints,
+            StreamWriteConstraints streamWriteConstraints,
+            BufferRecycler br, ContentReference contentRef, boolean managedResource,
             JsonEncoding enc)
     {
-        _streamReadConstraints = streamReadConstraints;
+        _streamReadConstraints = streamReadConstraints == null ?
```

Comment:
For 3.0 can also not allow `null`s I think so I'll change to `Objects.nonNull(...)` or such.